lms/static/js/spec/groups/views/cohorts_spec.js

```diff
@@ -0,0 +1,657 @@
+                    createCohortsView(this, {selectCohort: 1});
+
+                    addStudents(sixUsers);
+                    AjaxHelpers.expectRequest(requests, 'POST', '/mock_service/cohorts/1/add',
+                            'users=' + sixUsers.replace(/@/g, "%40").replace(/, /g, "%2C+")
+                    );
+                    respondToAdd({
+                        changed: [
+                            {email: 'moved1@sample.com', name: 'moved1', previous_cohort: 'group 2', username: 'moved1'},
+                            {email: 'moved2@sample.com', name: 'moved2', previous_cohort: 'group 2', username: 'moved2'},
+                            {email: 'moved3@sample.com', name: 'moved3', previous_cohort: 'group 3', username: 'moved3'}
+                        ],
+                        present: ['alreadypresent@sample.com']
+                    });
+                    respondToRefresh();
+
+                    verifyDetailedMessage('3 students have been added to this cohort group', 'confirmation',
+                        [
+                            "2 students were removed from group 2",
+                            "1 student was removed from group 3",
+                            "1 student was already in the cohort group"
+                        ]
+                    );
+                    expect(getStudentInput().val()).toBe('');
+                });
+
+                it('shows a message when the add fails', function() {
+                    createCohortsView(this, {selectCohort: 1});
+                    addStudents('student@sample.com');
+                    AjaxHelpers.respondWithError(requests);
+                    verifyMessage('Error adding students.', 'error');
+                    expect(getStudentInput().val()).toBe('student@sample.com');
+                });
+
+                it('clears an error message on subsequent add', function() {
+                    createCohortsView(this, {selectCohort: 1});
+
+                    // First verify that an error is shown
+                    addStudents('student@sample.com');
+                    AjaxHelpers.respondWithError(requests);
+                    verifyMessage('Error adding students.', 'error');
+
+                    // Now verify that the error is removed on a subsequent add
+                    addStudents('student@sample.com');
+                    respondToAdd({ added: ['student@sample.com'] });
+                    respondToRefresh(catLoversInitialCount + 1, dogLoversInitialCount);
+                    verifyMessage('1 student has been added to this cohort group', 'confirmation');
+                });
+            });
+
+            describe("Cohort Settings", function() {
+                describe("Content Group Setting", function() {
+                    it("shows a select element with an option for each content group", function () {
```

Comment:
I do this in the separate add tests: https://github.com/edx/edx-platform/blob/christina/link_group_content_to_cohort/lms/static/js/spec/groups/views/cohorts_spec.js#L265